provider/oracle/network/environ.go

```diff
@@ -152,8 +155,12 @@ func (e Environ) getSubnetInfo() ([]network.SubnetInfo, error) {
 
 // NetworkInterfaces is defined on the environs.Networking interface.
 func (e Environ) NetworkInterfaces(instId instance.Id) ([]network.InterfaceInfo, error) {
-	id := string(instId)
-	instance, err := e.client.InstanceDetails(id)
+	providerId, err := e.env.ProviderID(instId)
```

Comment:
Ah, right. How about this instead. Add the following method:

```go
func (env *OracleEnviron) OracleInstance(id instance.Id) (commonProvider.OracleInstance, error) {
    instances, err := env.getOracleInstances(id)
    if err != nil {
        return nil, errors.Trace(err)
    }
    return instances[0], nil
}
```

and the interface:

```go
package common

type OracleInstance interface {
    instance.Instance
    Details() response.Instance
}
```